zerver/views/realm.py

```diff
@@ -60,139 +73,118 @@ def parse_jitsi_server_url(
-        default=None,
-    ),
-    giphy_rating: Optional[int] = REQ(json_validator=check_int, default=None),
-    default_code_block_language: Optional[str] = REQ(default=None),
-    digest_weekday: Optional[int] = REQ(
-        json_validator=check_int_in(Realm.DIGEST_WEEKDAY_VALUES), default=None
-    ),
-    string_id: Optional[str] = REQ(
-        str_validator=check_capped_string(Realm.MAX_REALM_SUBDOMAIN_LENGTH),
-        default=None,
-    ),
-    org_type: Optional[int] = REQ(json_validator=check_int_in(Realm.ORG_TYPE_IDS), default=None),
-    enable_spectator_access: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    want_advertise_in_communities_directory: Optional[bool] = REQ(
-        json_validator=check_bool, default=None
-    ),
-    enable_read_receipts: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    move_messages_within_stream_limit_seconds_raw: Optional[Union[int, str]] = REQ(
-        "move_messages_within_stream_limit_seconds",
-        json_validator=check_string_or_int,
-        default=None,
-    ),
-    move_messages_between_streams_limit_seconds_raw: Optional[Union[int, str]] = REQ(
-        "move_messages_between_streams_limit_seconds",
-        json_validator=check_string_or_int,
-        default=None,
-    ),
-    enable_guest_user_indicator: Optional[bool] = REQ(json_validator=check_bool, default=None),
-    can_access_all_users_group_id: Optional[int] = REQ(
-        "can_access_all_users_group", json_validator=check_int, default=None
-    ),
+    new_stream_announcements_stream_id: Optional[Json[int]] = None,
+    signup_announcements_stream_id: Optional[Json[int]] = None,
+    zulip_update_announcements_stream_id: Optional[Json[int]] = None,
+    message_retention_days_raw: Annotated[
+        Optional[Json[Union[int, str]]], ApiParamConfig("message_retention_days")
+    ] = None,
+    send_welcome_emails: Optional[Json[bool]] = None,
+    digest_emails_enabled: Optional[Json[bool]] = None,
+    message_content_allowed_in_email_notifications: Optional[Json[bool]] = None,
+    bot_creation_policy: Optional[Json[BotCreationPolicyEnum]] = None,
+    create_public_stream_policy: Optional[Json[CommonPolicyEnum]] = None,
+    create_private_stream_policy: Optional[Json[CommonPolicyEnum]] = None,
+    create_web_public_stream_policy: Optional[Json[CreateWebPublicStreamPolicyEnum]] = None,
+    invite_to_stream_policy: Optional[Json[CommonPolicyEnum]] = None,
+    move_messages_between_streams_policy: Optional[
+        Json[MoveMessagesBetweenStreamsPolicyEnum]
+    ] = None,
+    user_group_edit_policy: Optional[Json[CommonPolicyEnum]] = None,
+    private_message_policy: Optional[Json[PrivateMessagePolicyEnum]] = None,
+    wildcard_mention_policy: Optional[Json[WildcardMentionPolicyEnum]] = None,
+    video_chat_provider: Optional[Json[int]] = None,
+    jitsi_server_url_raw: Annotated[
+        Optional[Json[str]],
+        AfterValidator(lambda val: check_jitsi_url(val)),
```

Comment:
`pydantic` has [URL types](https://docs.pydantic.dev/2.3/usage/types/urls/) but it has multiple available types and was not sure which will be same as `URLValidator` from django which we use currently, so did this to let pydantic only check string and then the url is checked by django validator.

Reply:
That sounds correct for this transition but is worth a #backend conversation about potential refactoring of this pattern.

Reply:
Started https://chat.zulip.org/#narrow/stream/3-backend/topic/validating.20URLs.20in.20typed_endpoint.20views